Feat/activity form/create and edit form activities

andreajhl opened the pull request:
-Add get, post and put functions in activityServer.
-Add activity creation and editing form, made with formik and ckeditor.
-Add modular function to render Formik form errors
-delete tasks not related to this branch
-Add modular function to get url of file type entries
-You don't have styles yet because you have to decide as a group.

Form:

-title: validation => long (minimum 4 letters)
-image: validation => must exist and accept jpg or png, render image over input on loading, as expected
-description: validation => must exist

![image](https://user-images.githubusercontent.com/75461111/139187024-f778a25a-d498-4cb5-905e-01acc24fa754.png)

Commit:
feat: :construction: agg request http (get,post,put) for form activity

Commit:
feat: :sparkles: add form for create and edit activities

Commit:
style: :lipstick: modify layout

Commit:
style: :lipstick: modification layout

saralavanchy commented on src/Components/Activities/ActivitiesForm.js:
```diff
@@ -1,32 +1,109 @@
+
+export default function ActivitiesForm({idActivity}){
+
+
```
this spaces are not necessary

saralavanchy commented on src/Components/Activities/ActivitiesForm.js:
```diff
@@ -1,32 +1,109 @@
+import { CKEditor } from '@ckeditor/ckeditor5-react';
+import {Formik, Form, Field,ErrorMessage} from 'formik';
+import ClassicEditor from '@ckeditor/ckeditor5-build-classic';
+
```
this space is not necessary

saralavanchy commented on src/Components/Activities/ActivitiesForm.js:
```diff
@@ -1,32 +1,109 @@
+
+import {getDataForm, postDataForm, putDataForm} from './function';
+
+export default function ActivitiesForm({idActivity}){
```
please, follow the convention of the proyect and use ```const ActivitiesForn = ({activityId}) => {}```
Also its more redeable activityId than idActivity.

saralavanchy commented on src/Components/Activities/ActivitiesForm.js:
```diff
@@ -1,32 +1,109 @@
+
+
+  const [imgState,setImg] = useState('');
+  const [state, setstate] = useState('');
```
please use setState instead of setstate, and also, what this state is for, what do it represent? the name is too generic

saralavanchy commented on src/Components/Activities/ActivitiesForm.js:
```diff
@@ -1,32 +1,109 @@
\ No newline at end of file
+      <div className='form-container'>
+          <Formik
+              initialValues={state? {
```
this validation is too log to be done here, please create a function that get the initial values.

saralavanchy commented on src/Components/Activities/ActivitiesForm.js:
```diff
@@ -1,32 +1,109 @@
+                  description:'',
+                  image:'',
+              }}
+              validate={(valores)=>{
```
Please create a validation function for this.

saralavanchy replied:
Also, if you are writting code on english it should be values instead of valores

saralavanchy commented on src/Components/Activities/ActivitiesForm.js:
```diff
@@ -1,32 +1,109 @@
+                  
+                  return errors
+              }}
+              onSubmit={(values)=>{
```
please create an onSubmit function to do this or a function whith a name that explains what is being done here.

saralavanchy commented on src/Components/Activities/ActivitiesForm.js:
```diff
@@ -1,32 +1,109 @@
+                            name="name"
+                            placeholder="Activity Name" 
+                          /> 
+                          <ErrorMessage name='name' component={()=><p className='form_div_error'>{errors.name}</p>}/>
```
please, this code is being repeted so create a function to show it,

saralavanchy replied:
Additionally, this code is also repeted on PR #4 so, maybe you can create a generic method on some type of util of components.

saralavanchy commented on src/Components/Activities/ActivitiesForm.js:
```diff
@@ -1,32 +1,109 @@
+                              config={ {
+                                  placeholder:"Write some activity description",
+                              } }
+                              onChange={( event, editor )=> {
```
create a function to do this

saralavanchy commented on src/Components/Activities/ActivitiesForm.js:
```diff
@@ -1,32 +1,109 @@
+                      <div >
+                          <p >Image</p>
+                          <div>
+                            {imgState.length>0 && <img src={imgState} alt='imagen vista previa'  width='180' height='180'  />} 
```
what means imgState.length>0 ? please create a constant of function to make this validation clearer

saralavanchy commented on src/Components/Activities/ActivitiesForm.js:
```diff
@@ -1,32 +1,109 @@
+                                id='image'                                
+                                name='image' 
+                                value={undefined}
+                                onChange={(e) => {
```
create a function to do this

saralavanchy commented on src/Components/Activities/function/index.js:
```diff
@@ -0,0 +1,18 @@
+import axios from "axios";
```
this look to be a service, so it should be on the services folder on a file with the service name that you are creating. I think, (for the name of the folder and the endpoints) that its the activities service, so, create a ActivitiesService with a findActivity, createActivity and modifyActivity functionallities

saralavanchy commented on src/Components/Projects/ProjectsForm.js:
```diff
@@ -1,32 +1,123 @@
-import React, { useState } from 'react';
-import '../FormStyles.css';
+import React, {useEffect, useState} from 'react';
```
this code is not scope of this PR, please remove it.

saralavanchy commented on src/Components/Projects/function/index.js:
```diff
@@ -0,0 +1,18 @@
+import axios from "axios";
+
```
this code is not scope of this PR, please remove it

saralavanchy reviewed (changes requested):
I left some comments but the code was a little bit difficult to follow, so, after this changes I will maybe give a better feedback. Thanks!

Commit:
fix: :adhesive_bandage: modularization of the activity form

it extracts in two functions the rendering of errors, and the obtaining of the url of the image inout

Commit:
feat: :sparkles: add request HTTP ( get, post, put) for the form activity

Commit:
fix: :rewind: revert the changes of the project form

reverts the changes of the project form to its original state and deletes the folder created because it is from another unrelated activity

Commit:
feat: :sparkles: add modularized functions to get the url of an image

add modularized functions to get the url of an image, and save it in a state

Commit:
feat: :sparkles: adds modularized functions to render form errors

adds modularized functions to render form errors made with formik, requires name and error

Commit:
fix: :adhesive_bandage: fix saved image data

fix the saved image data, now saving the 64bit url

Commit:
fix: :adhesive_bandage: fix saved image data

fix the saved image data, now saving the 64bit url

Commit:
delete console.log in disuse

Commit:
feat: :sparkles: create functions to validate formik form fields and return errors

It includes the replace function for the "<p></p>" of the ckditor component and a function to validate the values of the formik input

Commit:
feat: :sparkles: form modularization

extra the form and replace validations from ckeditor in a function in the common folder

Commit:
feat: :sparkles: modify variable names and separate functions in external folders

modify variable names to be more representative and extract functions in the common folder to make the component modularized and reusable

Commit:
feat: :fire: delete unused files

the functions in these files were moved to new files in their corresponding folders

Commit:
feat: :sparkles: add function to render form errors made with formik

Commit:
feat: :sparkles: add function to read and set url of input file

add function to read and set its value through a callback function

Commit:
feat: :sparkles: add function to remove the "<p> </p>" from the value of ckeditor

Commit:
feat: :sparkles: add function to validate the form fields and return the corresponding errors

Commit:
feat: :fire: delete unused files
the functions in these files were moved to new files in their corresponding folders

Commit:
feat: :sparkles: modify variable names and separate functions in external folders
modify variable names to be more representative and extract functions in the common folder to make the component modularized and reusable

Commit:
feat: :sparkles: modify variable name

Commit:
feat: :sparkles: add function to dispatch the creation or modification request depending on whether there is or an id

Commit:
Merge branch development into feat/activity-Form/create-and-edit-form-activities

Commit:
feat: :sparkles: modified name route for import set UrlImage

Commit:
feat: :sparkles: modified route request api rest

Commit:
feat: :sparkles: add route in APP for edit activity

Commit:
feat: :sparkles: modifies the initial value of the form momentarily, until you have redux

Commit:
feat: :sparkles: validation to decide which request to make

Commit:
feat: :sparkles: validation to decide which request to make

Commit:
feat: :fire: remove import and delete file unused to counter

Commit:
Merge branch development into feat/activity-Form/create-and-edit-form-activities

andreajhl committed:
Merge branch 'development' into feat/activity-Form/create-and-edit-form-activities

andreajhl committed:
Merge branch 'development' into feat/activity-Form/create-and-edit-form-activities

andreajhl committed:
Merge branch 'development' into feat/activity-Form/create-and-edit-form-activities
